### DISCUSSION TITLE
Restore sessions on backend

### DISCUSSION THREAD

**Discussion by ScorpioEF** [WalletConnect/walletconnect-monorepo#2511] [Developer Support] [answered]
Hi everyone,

We are currently using wallet connect v1 on backend with node.js.

I'm currently working on migrating the library from v1 to v2 and I'm facing an issue regarding session saving and restoring when the server restarts. In v1, we were using the `session` property by serializing it to save it and injecting it in the constructor to restore it.

I tried to do the same thing with the `sessions` in the sign client, but it doesn't seem to work. Is there a way to do it?

**Comment by finessevanes** [comment]
Hey - I think these are the docs you're looking for: https://docs.walletconnect.com/2.0/web/sign/dapp-usage#restoring-a-session

**Reply by ScorpioEF** [reply]
Hey, thanks for your answer. Our goal was to handle everything server side so we don't really have access to the browser local storage.

With V1, we were serializing the `session` property of the connector, storing it in database. If the server were to restart, we could take it and inject it in the constructor of the connector.

**Reply by ScorpioEF** [reply, marked as answer]
Ok, found a solution. You can implement your own storage solution in Core API.

**Reply by arbaz-meo** [reply]
@ScorpioEF can you explain little bit more for me how you implemented this
